internal/api/option/auth.go

```diff
@@ -250,18 +250,22 @@ func (a *authInterceptor) listServiceAccounts(
 	ctx context.Context,
 	c claims,
 ) (map[string]map[types.NamespacedName]struct{}, error) {
-	queries := []libClient.MatchingFields{
-		{
-			kubeclient.ServiceAccountsByOIDCSubjectIndexField: c.Subject,
-		},
-		{
-			kubeclient.ServiceAccountsByOIDCEmailIndexField: c.Email,
-		},
-	}
-	for _, group := range c.Groups {
-		queries = append(queries, libClient.MatchingFields{
-			kubeclient.ServiceAccountsByOIDCGroupIndexField: group,
-		})
+	queries := []libClient.MatchingFields{}
+	for _, claim := range c {
+		if claimString, ok := claim.(string); ok {
+			queries = append(queries, libClient.MatchingFields{
+				kubeclient.ServiceAccountsByOIDCClaimIndexField: claimString,
+			})
+		}
+		if claimSlice, ok := claim.([]any); ok {
+			for _, claimSliceItem := range claimSlice {
+				if claimSliceItemString, ok := claimSliceItem.(string); ok {
+					queries = append(queries, libClient.MatchingFields{
+						kubeclient.ServiceAccountsByOIDCClaimIndexField: claimSliceItemString,
+					})
+				}
+			}
+		}
 	}
```

Comment:
These queries don't look right. I think the index is wrong also.

The problem is that we're saying, "look for a ServiceAccount with a claim with a certain value," but we're not taking the _kind_ of claim into account. As it stands now, a user could have sub foo and we would consider them having access to to that SA if the SA had _group_ foo.

I think that when the index is built, the values returned need to be of the form "<claim-kind>/<claim-value>" and when searching the index, input needs to take the same form.

Reply:
I think I understand the part about not including the claim name as well as the value could result in finding SAs that it shouldn't. 

I'm a bit confused about the last part about being in the form "/". Do you mean that if we want to include the claim name and value it needs to be separated by "/" e.g. group/foo, sub/foo?

Reply:
Oh wow. Sorry. Formatting issue because I used angled brackets and GitHub's markdown renderer interpreted them as invalid HTML. You correctly interpreted my intent though. `<claim name>/<claim value>` or something along those lines. `<claim name=claim value>`? You get the idea.

Reply:
Yeah that makes sense, I've modified the indexer to look for  "<claimName>/<claimValue>".